llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1258,6 +1258,36 @@ instCombineSVEST1(InstCombiner &IC, IntrinsicInst &II, const DataLayout &DL) {
   return IC.eraseInstFromFunction(II);
 }
 
+static std::optional<Instruction *>
+instCombineSVESplice(InstCombiner &IC, IntrinsicInst &II, const DataLayout &DL,
+                     const AArch64Subtarget *ST) {
+  Value *Pred = II.getOperand(0);
+  Value *VecOp0 = II.getOperand(1);
+  Value *VecOp1 = II.getOperand(2);
+  Value *PtrOp;
+  unsigned MinSVESize = ST->getMinSVEVectorSizeInBits();
+  unsigned MaxSVESize = ST->getMaxSVEVectorSizeInBits();
```

Comment:
These functions shouldn't really be used outside of code generation.  I think instead you actually want to query the `vscale_range` function attributes.

Reply:
Thanks @paulwalker-arm , update with API `getVScaleRange` to check function attribute.